src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +451,74 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
+    {
+        isParameterlessConstructor = null;
+        matchedParameters = null;
+        if (type.IsAbstract)
+        {
+            return false;
+        }
+
+        var constructors = type.Constructors.Where(constructor => constructor.DeclaredAccessibility == Accessibility.Public && !constructor.IsStatic);
+        var numOfConstructors = constructors.Count();
+        // When leveraging parameterless constructors, we want to ensure we only emit for writable
+        // properties. We do not have this constraint if we are leveraging a parameterized constructor.
+        var properties = type.GetMembers().OfType<IPropertySymbol>().Where(property => property.DeclaredAccessibility == Accessibility.Public);
+        var writableProperties = properties.Where(property => !property.IsReadOnly);
+
+        if (numOfConstructors == 1)
+        {
+            var targetConstructor = constructors.SingleOrDefault();
+            var lookupTable = new Dictionary<ParameterLookupKey, IPropertySymbol>();
+            foreach (var property in properties)
+            {
+                lookupTable.Add(new ParameterLookupKey(property.Name, property.Type), property);
+            }
+
+            var parameters = targetConstructor.GetParameters();
+            var propertiesWithParameterInfo = new List<ConstructorParameter>();
+
+            if (parameters.Length == 0)
+            {
+                isParameterlessConstructor = true;
+                matchedParameters = writableProperties.Select(property => new ConstructorParameter(property, null));
+                return true;
+            }
+
+            foreach (var parameter in parameters)
+            {
+                var key = new ParameterLookupKey(parameter.Name!, parameter.Type);
+                if (lookupTable.TryGetValue(key, out var property))
+                {
+                    propertiesWithParameterInfo.Add(new ConstructorParameter(property, parameter));
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            isParameterlessConstructor = false;
+            matchedParameters = propertiesWithParameterInfo;
+            return true;
+        }
+
+        if (type.IsValueType)
+        {
+            isParameterlessConstructor = true;
```

Comment:
I guess this is why, but I'm still a bit confused - parameterless is true and there are a non-zero number of parameters?

Reply:
Yeah, the term parameters is overloaded here. This is referring to the parameterless constructor invoked with object initializers for the properties defined in `matchedParameters`.

When you're using the default parameterless constructor, any writable property can be used in the object initialization.

When you're using a defined constructor, only the writable properties that have a match in the constructor signature can be used.

I think a clearer set of names for this might be `useDefaultConstructor` and `matchedProperties`.

Reply:
Yeah, renaming seems like it would help.